bazel/EXTERNAL_DEPS.md

```diff
@@ -1,4 +1,37 @@
-# Adding external dependencies to Envoy
+# Adding external dependencies to Envoy (native Bazel)
+
+This is the preferred style of adding dependencies that use Bazel for their
+build process.
+
+1. Specify name and version in [external documentation](../docs/install/requirements.rst).
+2. Define a new Bazel repository in [`bazel/repositories.bzl`](repositories.bzl),
+   in the `envoy_dependencies()` function.
+5. Reference your new external dependency in some `envoy_cc_library` via the
+   `external_deps` attribute.
+6. `bazel test //test/...`
+
+# Adding external dependencies to Envoy (genrule repository)
+
+This is the newer style of adding dependencies with no upstream Bazel configs.
+It wraps the dependency's native build tooling in a Bazel genrule, installing
+to a Bazel-managed prefix.
+
+1. Specify name and version in [external documentation](../docs/install/requirements.rst).
+2. Add a BUILD file in [`bazel/external/`](external/), using a genrule to build
+   the dependency. Please do not add BUILD logic that replaces the dependency's
+   upstream build tooling.
+2. Define a new Bazel repository in [`bazel/repositories.bzl`](repositories.bzl),
+   in the `envoy_dependencies()` function. The repository may use `genrule_repository`
+   from [`bazel/genrule_repository.bzl`](genrule_repository.bzl) to place large
+   genrule commands into a separate file.
```

Comment:
This is pretty complicated for the average Envoy developer who isn't a Bazel expert. A build recipe was easy to understand, it's just a shell script. Can you provide some simple explanations above imagining the developer has never heard of a `genrule`? Even a link to Bazel genrule stuff would help, but we don't want folks to have to become Bazel ninjas to add deps.

Reply:
Made a few tweaks, does this help? I added a link to the genrule docs, and phrased it as a "bash command with Bazel extensions" which should hopefully be less intimidating to devs unfamiliar with Bazel.